### ISSUE TITLE
target version@3.15: `tomllib` treated as non-standard library; is this an expected change?

### ISSUE THREAD

**Issue by k0pernikus** [astral-sh/ruff#25100]
### Question

In a project of mine I upgraded from python 3.14 to 3.15.0b1.

I also changed the target in `ruff.toml`

target-version = "py315"

These are my rules, I haven't change anything there for the python upgrade:

```toml

target-version = "py315"
line-length = 180

[lint]
select = [
  "F",   # Pyflakes
  "E",   # Pycodestyle Error
  "W",   # Pycodestyle Warning
  "UP",  # Pyupgrade
  "B",   # Bugbear
  "SIM", # Simplify
  "I",   # Isort
  "RET", # Return (Guard clauses)
  "FBT", # Boolean Trap
  "C4",  # Comprehensions
  # "ANN",  # Annotations (No Any)
  "ERA",   # Eradicate (Commented-out code)
  "PIE",   # Pie
  "PLR",   # Pylint Refactor
  "PLE",   # Pylint Error
  "EM",    # Error Message (Enforce variables for strings),
  "TD006",
]

ignore = [
  "ARG",     # creates too many false positive :( (esp. on method overwrites)
  "D",       # No docstrings
  "RET504",  # Allow variable assignment before return (Debuggable)
  "B028",    # changes code behavior
  "FBT001",  # might be a bit too pedantic
  "PLR0913", # I support the idea but dealing with playwright does make it usable
  "ERA001",  # can be meaningful to document intent
  "EM102",
  "EM101",   # too verbose, esp. with AI code generation
  "PLR2004", # too strict
]

# see: https://github.com/astral-sh/ruff/issues/22620#issue-3821861117
extend-unsafe-fixes = [
  "F541", # f"Hello {name}" intended, but f"Hello name" written, would be "fixed" into "Hello name"
]

[lint.isort]
combine-as-imports = true
lines-after-imports = 2

[lint.flake8-annotations]
# Forbid 'Any' for regular args (ANN401), but allow for variadic pass-throughs
allow-star-arg-any = true

[lint.flake8-unused-arguments]
# Support the underscore approach for forced API/Inheritance arguments
ignore-variadic-names = true

[lint.flake8-bugbear]
extend-immutable-calls = [
  "fastapi.Depends",
  "fastapi.Query",
  "click.option",
]

[format]
quote-style = "double"
indent-style = "space"
skip-magic-trailing-comma = false
```

Now it warns in my code base:

```shell
just check
uv lock --check
Resolved 81 packages in 5ms
ruff check .
warning: Support for Python 3.15 is under development and may be unstable. Enable `preview` to remove this warning.
I001 [*] Import block is un-sorted or un-formatted
  --> cli/src/single_test/collector.py:1:1
   |
 1 | / import os
 2 | | import re
 3 | | import subprocess
 4 | | import sys
 5 | | import tomllib
 6 | | from collections.abc import Iterator
 7 | | from concurrent.futures import ThreadPoolExecutor, as_completed
 8 | | from pathlib import Path
 9 | |
10 | | from single_test.packages import load_packages
   | |______________________________________________^
   |
help: Organize imports

Found 1 error.
[*] 1 fixable with the `--fix` option.
error: recipe `check` failed on line 39 with exit code 1
```

and wants to fix it as such:

```python
import os
import re
import subprocess
import sys
from collections.abc import Iterator
from concurrent.futures import ThreadPoolExecutor, as_completed
from pathlib import Path

import tomllib

from single_test.packages import load_packages


_TEST_ID_RE = re.compile(r"^\S+::\S")
...
```

Is it wanted behavior to have tomlib in its own import block? IIANM, only non-standard libraries should have their own import section, no?

Is this a bug? Did I configure ruff incorrectly for python 3.15?

The code linted before (target 3.14) just wine with one sorted import block like so:

```python
import os
import re
import subprocess
import sys
import tomllib
from collections.abc import Iterator
from concurrent.futures import ThreadPoolExecutor, as_completed
from pathlib import Path

from single_test.packages import load_packages
...
```

### Version

ruff 0.15.12

**Comment by k0pernikus** [comment]
I realized there was a warning:

```
warning: Support for Python 3.15 is under development and may be unstable. Enable `preview` to remove this warning.
```

Yet enabling this does not only make the warning disappear, it actually show a lot more linting issues. This is a bit unexpected due to the wording but makes sense.

So if I understand this correctly:

Changing the target version of course would introduce rule changes, e..g. from 3.14 to 3.15. I just got lucky that only one rule changed.

Enabling preview will furthermore load my configured rules in their most up to date state, making violations more likely.

Under that assumption, it would be expected to see breaking changes by bumping the target version and by enabling the preview.

My biggest confusion stemmed from the fact that the import change was the only complaint after the upgrade, if I saw all that I also saw with preview enabled, I would not even have question its change.

I just changed my ruff.toml for my python@3.14 sate and rerun uv check different ways:



---

```shell
uvx ruff check . --statistics --quiet --target-version py314 --no-preview
[all pass]
```

```shell
uvx ruff check . --statistics --quiet --target-version py314 --preview
3       PLR0917 too-many-positional-arguments
2       PLR6301 no-self-use
1       B903    class-as-data-structure
1       PLR6201 literal-membership
```

```shell
uvx ruff check . --statistics --quiet --target-version py315 --no-preview
1       I001    [*] unsorted-imports
```

```shell
uvx ruff check . --statistics --quiet --target-version py315 --preview
3       PLR0917 [ ] too-many-positional-arguments
2       PLR6301 [ ] no-self-use
1       B903    [ ] class-as-data-structure
1       I001    [*] unsorted-imports
1       PLR6201 [ ] literal-membership
```

Still interesting to see that the `unsorted-imports` seems to be exclusive to the 3.15 target. I would have expected that to also pop up for 3.14 with preview enabled. So I assume it is a bug that it is treated as non-default library.

If that is the case I could do for now

```toml
[lint.isort]
extra-standard-library = ["tomllib"]
combine-as-imports = true
lines-after-imports = 2
```
yet since it is autofixable, I will use whatever ruff wants :)

**Comment by MichaReiser** [comment]
This is https://github.com/astral-sh/ruff/issues/21119. Ruff doesn't fully support Python 3.15 yet, but I think we're at a point where we could generate the Python 3.15 stubs and later update them before the release
